GenericFlag doesn't work with Destination: type *Generic is pointer to interface, not interface

## My urfave/cli version is

v2.11.1

## Checklist

- [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
- [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/main/docs/v2/manual.md)
- [x] Did you perform a search about this problem? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## Dependency Management

<!--
  Delete any of the following that do not apply:
-->

- My project is using go modules.

## Describe the bug

When I configure a `GenericFlag` with a `Destination` the code doesn't compile.

## To reproduce

1. In the https://github.com/urfave/cli repository, paste the code below into the `flag_test.go` file.

```go
type genericType struct {
	s string
}

func (g *genericType) Set(value string) error {
	g.s = value
	return nil
}

func (g *genericType) String() string {
	return g.s
}

func TestParseDestinationGeneric(t *testing.T) {
	var dest *genericType
	_ = (&App{
		Flags: []Flag{
			&GenericFlag{
				Name:        "dest",
				Destination: dest,
			},
		},
		Action: func(ctx *Context) error {
			if dest.String() != "10,20" {
				t.Errorf("expected destination Generic 10,20")
			}
			return nil
		},
	}).Run([]string{"run", "--dest", "10,20"})
}
```

2. Run `go test .` command.

## Observed behavior

```
./flag_test.go:2187:18: cannot use dest (variable of type *genericType) as type *Generic in struct literal:
        *genericType does not implement *Generic (type *Generic is pointer to interface, not interface)
```

## Expected behavior

The value should be set in the destination.

## Want to fix this yourself?

Please see https://github.com/urfave/cli/pull/1442

## Run `go version` and paste its output here

```
go version go1.18.3 darwin/amd64
```

--- a/cmd/urfave-cli-genflags/generated.gotmpl
+++ b/cmd/urfave-cli-genflags/generated.gotmpl
@@ -17,7 +17,7 @@ type {{.TypeName}} struct {
   HasBeenSet bool
 
   Value {{if .ValuePointer}}*{{end}}{{.GoType}}
-  Destination *{{.GoType}}
+  Destination {{if .NoDestinationPointer}}{{else}}*{{end}}{{.GoType}}
 
   Aliases []string
   EnvVars []string

--- a/cmd/urfave-cli-genflags/main.go
+++ b/cmd/urfave-cli-genflags/main.go
@@ -223,10 +223,11 @@ func (gfs *Spec) SortedFlagTypes() []*FlagType {
 }
 
 type FlagTypeConfig struct {
-	SkipInterfaces []string           `yaml:"skip_interfaces"`
-	StructFields   []*FlagStructField `yaml:"struct_fields"`
-	TypeName       string             `yaml:"type_name"`
-	ValuePointer   bool               `yaml:"value_pointer"`
+	SkipInterfaces       []string           `yaml:"skip_interfaces"`
+	StructFields         []*FlagStructField `yaml:"struct_fields"`
+	TypeName             string             `yaml:"type_name"`
+	ValuePointer         bool               `yaml:"value_pointer"`
+	NoDestinationPointer bool               `yaml:"no_destination_pointer"`
 }
 
 type FlagStructField struct {
@@ -254,6 +255,14 @@ func (ft *FlagType) ValuePointer() bool {
 	}
 
 	return ft.Config.ValuePointer
+}
+
+func (ft *FlagType) NoDestinationPointer() bool {
+	if ft.Config == nil {
+		return false
+	}
+
+	return ft.Config.NoDestinationPointer
 }
 
 func (ft *FlagType) TypeName() string {

--- a/flag-spec.yaml
+++ b/flag-spec.yaml
@@ -5,62 +5,71 @@
 flag_types:
   bool: 
     struct_fields:
-      - { name: Count, type: int, pointer: true }
-  float64: {}
-  int64:
-    struct_fields:
-      - { name: Base, type: int }
-  int:
-    struct_fields:
-      - { name: Base, type: int }
-  time.Duration: {}
-  uint64:
-    struct_fields:
-      - { name: Base, type: int }
-  uint:
-    struct_fields:
-      - { name: Base, type: int }
-
-  string:
-    struct_fields:
-      - { name: TakesFile, type: bool }
-  Generic:
-    struct_fields:
-      - { name: TakesFile, type: bool }
-  Path:
-    struct_fields:
-      - { name: TakesFile, type: bool }
-
+      - name: Count 
+        type: int
+        pointer: true
+  float64: 
   Float64Slice:
     value_pointer: true
     skip_interfaces:
       - fmt.Stringer
-  Int64Slice:
-    value_pointer: true
-    skip_interfaces:
-      - fmt.Stringer
+  int:
+    struct_fields:
+      - name: Base
+        type: int
   IntSlice:
     value_pointer: true
     skip_interfaces:
       - fmt.Stringer
+  int64:
+    struct_fields:
+      - name: Base
+        type: int
+  Int64Slice:
+    value_pointer: true
+    skip_interfaces:
+      - fmt.Stringer
+  uint:
+    struct_fields:
+      - name: Base
+        type: int
+  UintSlice: 
+    value_pointer: true
+    skip_interfaces:
+      - fmt.Stringer
+  uint64:
+    struct_fields:
+      - name: Base
+        type: int
+  Uint64Slice: 
+    value_pointer: true
+    skip_interfaces:
+      - fmt.Stringer        
+  string:
+    struct_fields:
+      - name: TakesFile
+        type: bool
   StringSlice:
     value_pointer: true
     skip_interfaces:
       - fmt.Stringer
     struct_fields:
-      - { name: TakesFile, type: bool }
+      - name: TakesFile
+        type: bool
+  time.Duration:      
   Timestamp:
     value_pointer: true
     struct_fields:
-      - { name: Layout, type: string }
-      - { name: Timezone, type: "*time.Location" }
-
-  UintSlice: 
-    value_pointer: true
-    skip_interfaces:
-      - fmt.Stringer
-  Uint64Slice: 
-    value_pointer: true
-    skip_interfaces:
-      - fmt.Stringer
-
+      - name: Layout 
+        type: string 
+      - name: Timezone
+        type: "*time.Location"
+  Generic:
+    no_destination_pointer: true
+    struct_fields:
+      - name: TakesFile
+        type: bool
+  Path:
+    struct_fields:
+      - name: TakesFile
+        type: bool

--- a/godoc-current.txt
+++ b/godoc-current.txt
@@ -1085,7 +1085,7 @@ type GenericFlag struct {
 	HasBeenSet bool
 
 	Value       Generic
-	Destination *Generic
+	Destination Generic
 
 	Aliases []string
 	EnvVars []string

--- a/zz_generated.flags.go
+++ b/zz_generated.flags.go
@@ -58,7 +58,7 @@ type GenericFlag struct {
 	HasBeenSet bool
 
 	Value       Generic
-	Destination *Generic
+	Destination Generic
 
 	Aliases []string
 	EnvVars []string

